### DISCUSSION TITLE
Inquiry Regarding Phalcon Framework Support for PHP 8.2

### DISCUSSION THREAD

**Discussion by prachipachpande** [phalcon/cphalcon#16487] [General]
I am a developer who greatly values the functionalities and capabilities offered by the Phalcon framework.
I am writing to inquire about the anticipated release date for the Phalcon version that supports PHP 8.2. As per Phalcon documentation information provided by your team, it was mentioned that the version accommodating PHP 8.2 was expected to be released in September 2023. However, to our knowledge, this version has not yet been made available.
 
Our team is eagerly anticipating this release to leverage the benefits and enhancements offered by both Phalcon and PHP 8.2 for our ongoing projects. Having an estimated timeline for the release would immensely help us plan our development and upgrade strategies accordingly.

Could you kindly provide any updates or insights regarding the expected release date of the Phalcon version compatible with PHP 8.2? We understand that development timelines can fluctuate, but having a rough estimate would greatly assist us in our planning process.

We highly value the contributions and efforts of the Phalcon team in delivering exceptional software, and we appreciate any information you can provide regarding this matter.

Thank you for your attention to this request. We eagerly await your response.

**Comment by Jeckerson** [comment]
Hello, PHP 8.2 is supported by latests versions of Phalcon, there are some warnings due dynamic properties, which might be disabled via ini value or by `#[AllowDynamicProperties]` attribute.

These bugs are known and we are still working on them. But in general you can use PHP 8.2 and in case of some bugs - please submit an issue here.

**Reply by prachipachpande** [reply]
Hello, I wanted to bring to your attention an issue I encountered while attempting to install the latest version of Phalcon in PHP 8.2. After completing the installation process as per the documentation, Phalcon does not appear to be reflected in the phpinfo file of PHP. I'm using windows server. Could you please provide guidance or support in troubleshooting this matter?

**Reply by stfast** [reply]
Check the module existance:

$ php -m |grep phalcon
phalcon

if it is not there, than you shoud add extension to php.ini file

$ cat /etc/php.d/99-phalcon.ini 
extension=phalcon.so

**Reply by prachipachpande** [reply]
Hello, I'm using Windows. 
Following the Phalcon documentation, I've completed the installation steps, including:

1. Downloading the appropriate Phalcon DLL file compatible with PHP 8.2 (Phalcon Version | 5.1.4 )
2. Adding the extension to the PHP configuration file (php.ini) using the line: extension=php_phalcon.dll
3. Ensuring that the php_phalcon.dll file is in the correct extension directory and confirming the correct path in the configuration
4. Restarting the web server to apply the changes.

 I've thoroughly reviewed the PHP configuration and restarted the server multiple times, yet the issue persists.

**Reply by stfast** [reply]
Yes, I know you are using windows, still PHP basics are the same.

Is there an error when you run PHP in CMD:

$ C:\path\to\php -m

Sometimes it happens, that  PHP module load order (in php.ini or php.d/XX-module.ini) should be right, 
before or after some other PHP module.

**Comment by prachipachpande** [comment]
I am reaching out to report an issue I encountered while attempting to integrate Phalcon 8.2 framework version with XAMPP 8.2. Below is the error message I received:
PHP Warning: PHP Startup: Unable to load dynamic library 'php_phalcon.dll' (tried: \\xampp\\php\\ext\\php_phalcon.dll (The specified module could not be found), \\xampp\\php\\ext\\php_php_phalcon.dll.dll (The specified module could not be found)) in Unknown on line 0
PHP Warning: PHP Startup: psr: Unable to initialize module\nModule compiled with module API=20210902\nPHP compiled with module API=20220829\nThese options need to match\n in Unknown on line 0

Here are the details of my environment:

Phalcon Version: 8.2
XAMPP Version: 8.2
Operating System: windows 64-bit 

I have tried the following troubleshooting steps:

1.Checked compatibility between Phalcon and PHP versions.
2.Verified that the correct extension file (php_phalcon.dll) is present in the specified directory.

Despite these efforts, I am still unable to resolve the issue. Your assistance in resolving this matter would be greatly appreciated.
